src/daft-parquet/src/metadata.rs

```diff
@@ -12,11 +16,220 @@ fn metadata_len(buffer: &[u8], len: usize) -> i32 {
     i32::from_le_bytes(buffer[len - 8..len - 4].try_into().unwrap())
 }
 
+struct ParquetTypeWrapper(ParquetType);
+
+impl TreeNode for ParquetTypeWrapper {
+    fn apply_children<F>(&self, op: &mut F) -> DaftResult<VisitRecursion>
+    where
+        F: FnMut(&Self) -> DaftResult<VisitRecursion>,
+    {
+        match &self.0 {
+            ParquetType::PrimitiveType(..) => Ok(VisitRecursion::Skip),
+            ParquetType::GroupType { fields, .. } => {
+                for child in fields.iter() {
+                    // TODO: Expensive clone here because of ParquetTypeWrapper type, can we get rid of this?
+                    match op(&ParquetTypeWrapper(child.clone()))? {
```

Comment:
Not sure if getting rid of this clone is possible with the wrapper struct approach. Also, `apply_children` isn't actually called in this PR since we only use the `mutate` functionality on `TreeNodeRewriter`.